src/dev/impl/DevToys/ViewModels/Tools/Text/StringUtilities/StringUtilitiesToolViewModel.cs

```diff
@@ -626,6 +634,182 @@ private void ExecuteInverseCaseCommand()
+
+        private void ExecuteAlphabetizeCommand()
+        {
+            string? text = _textBackup;
+            if (text is null)
+            {
+                return;
+            }
+
+            IOrderedEnumerable<string> lines = text.Split('\r').OrderBy(line => line, StringComparer.CurrentCulture);
+            text = string.Join('\r', lines);
+
+            lock (_lockObject)
+            {
+                _forbidBackup = true;
+                Text = text;
+                _forbidBackup = false;
+            }
+        }
+
+        #endregion
+
+        #region ReverseAlphabetizeCommand
+
+        public IRelayCommand ReverseAlphabetizeCommand { get; }
+
+        private void ExecuteReverseAlphabetizeCommand()
+        {
+            string? text = _textBackup;
+            if (text is null)
+            {
+                return;
+            }
+
+            IOrderedEnumerable<string> lines = text.Split('\r').OrderByDescending(line => line, StringComparer.CurrentCulture);
+            text = string.Join('\r', lines);
+
+            lock (_lockObject)
+            {
+                _forbidBackup = true;
+                Text = text;
+                _forbidBackup = false;
+            }
+        }
+
+        #endregion
+
+        #region AlphabetizeLastCommand
+
+        public IRelayCommand AlphabetizeLastCommand { get; }
+
+        private void ExecuteAlphabetizeLastCommand()
+        {
+            string? text = _textBackup;
+            if (text is null)
+            {
+                return;
+            }
+
+            var lines = text.Split('\r').ToList();
+            lines.Sort((line1, line2) =>
+            {
+                string? line1LastWord = line1.Split(' ', ',').LastOrDefault();
+                string? line2LastWord = line2.Split(' ', ',').LastOrDefault();
+
+                if (string.Equals(line1LastWord, line2LastWord, StringComparison.CurrentCulture))
+                {
+                    return 0;
+                }
+
+                var newList = new List<string> { line1LastWord, line2LastWord }.OrderBy(word => word).ToList();
+                return newList.IndexOf(line1LastWord) == 0 ? -1 : 1;
+            });
+            text = string.Join('\r', lines);
+
+            lock (_lockObject)
+            {
+                _forbidBackup = true;
+                Text = text;
+                _forbidBackup = false;
+            }
+        }
+
+        #endregion
+
+        #region ReverseAlphabetizeLastCommand
+
+        public IRelayCommand ReverseAlphabetizeLastCommand { get; }
+
+        private void ExecuteReverseAlphabetizeLastCommand()
+        {
+            string? text = _textBackup;
+            if (text is null)
+            {
+                return;
+            }
+
+            var lines = text.Split('\r').ToList();
+            lines.Sort((line1, line2) =>
+            {
+                string? line1LastWord = line1.Split(' ', ',').LastOrDefault();
```

Comment:
`Split` method allocates N string (and so memory). I feel like this detection could be done without allocating extra strings (except the one we're interested in).

How about reading the string from its end to its beginning and detect word delimiter characters (any non-alphanumeric character?) ?

Reply:
Sure, hopefully this does the trick: 429ffc252b67ac554fef2bdb546c0f41fdb97866